src/swappers/AuctionSwapper.sol

```diff
@@ -0,0 +1,216 @@
+ *   contract for the specific auction contract with the ability to override
+ *   any of the functions to implement custom hooks.
+ *
+ *   NOTE: If any hooks are not desired, the strategist should also
+ *   implement a way to call the {setHookFlags} on the auction contract
+ *   to avoid unnecessary gas for unused functions.
+ */
+contract AuctionSwapper {
+    using SafeERC20 for ERC20;
+
+    modifier onlyAuction() {
+        _isAuction();
+        _;
+    }
+
+    /**
+     * @dev Check the caller is the auction contract for hooks.
+     */
+    function _isAuction() internal view virtual {
+        require(msg.sender == auction, "!auction");
+    }
+
+    /// @notice The pre-deployed Auction factory for cloning.
+    // TODO: Update to real address
+    address public constant auctionFactory =
+        0x5991A2dF15A8F6A256D3Ec51E99254Cd3fb576A9;
+
+    /// @notice Address of the specific Auction this strategy uses.
+    address public auction;
+
+    /*//////////////////////////////////////////////////////////////
+                    AUCTION STARTING AND STOPPING
+    //////////////////////////////////////////////////////////////*/
+
+    function _enableAuction(
+        address _from,
+        address _want
+    ) internal virtual returns (bytes32) {
+        return _enableAuction(_from, _want, 1 days, 3 days, 1e6);
+    }
+
+    /**
+     * @dev Used to enable a new Auction to sell `_from` to `_want`.
+     *   If this is the first auction enabled it will deploy a new `auction`
+     *   contract to use from the factory.
+     *
+     * NOTE: This only supports one `_want` token per strategy.
+     *
+     * @param _from Token to sell
+     * @param _want Token to buy.
+     * @return .The auction ID.
+     */
+    function _enableAuction(
+        address _from,
+        address _want,
+        uint256 _auctionLength,
+        uint256 _auctionCooldown,
+        uint256 _startingPrice
+    ) internal virtual returns (bytes32) {
+        address _auction = auction;
+
+        // If this is the first auction.
+        if (_auction == address(0)) {
+            // Deploy a new auction
+            _auction = AuctionFactory(auctionFactory).createNewAuction(
+                _want,
+                address(this),
+                address(this),
+                _auctionLength,
+                _auctionCooldown,
+                _startingPrice
+            );
+            // Store it for future use.
+            auction = _auction;
+        } else {
+            // Can only use one `want` per auction contract.
+            require(Auction(_auction).want() == _want, "wrong want");
+        }
+
+        // Enable new auction for `_from` token.
+        return Auction(_auction).enable(_from);
+    }
+
+    /**
+     * @dev Disable an auction for a given token.
+     * @param _from The token that was being sold.
+     */
+    function _disableAuction(address _from) internal virtual {
+        Auction(auction).disable(_from);
```

Comment:
require auction != zero address?

Reply:
will revert during the auction.disable() call